pkg/mutation/system.go

```diff
@@ -9,29 +9,36 @@ import (
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/uuid"
 	"github.com/open-policy-agent/gatekeeper/pkg/logging"
+	"github.com/open-policy-agent/gatekeeper/pkg/mutation/reporter"
 	"github.com/open-policy-agent/gatekeeper/pkg/mutation/schema"
 	"github.com/open-policy-agent/gatekeeper/pkg/mutation/types"
 	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
-// System keeps the list of mutations and
-// provides an interface to apply mutations.
+// System keeps the list of mutators and provides an interface to apply mutations.
 type System struct {
 	schemaDB        schema.DB
 	orderedMutators []types.Mutator
 	mutatorsMap     map[types.ID]types.Mutator
 	mux             sync.RWMutex
+	reporter        *reporter.Reporter
 }
 
 // NewSystem initializes an empty mutation system.
-func NewSystem() *System {
+func NewSystem() (*System, error) {
```

Comment:
We should handle the reporter as an injected dependency.

This means that we'll need to either handle the nil case or provide a default, dummy reporter. On the plus side, that means we still don't need to check errors when creating a new mutation system.